politeiawww/user/mysql/mysql.go

```diff
@@ -437,13 +454,68 @@ func (m *mysql) UserUpdate(u user.User) error {
 		Blob:      eb,
 		UpdatedAt: time.Now().Unix(),
 	}
-	_, err = m.userDB.ExecContext(ctx,
+	_, err = tx.ExecContext(ctx,
 		"UPDATE users SET username = ?, uBlob = ?, updatedAt = ? WHERE ID = ? ",
 		ur.Username, ur.Blob, ur.UpdatedAt, ur.ID)
 	if err != nil {
 		return fmt.Errorf("create user: %v", err)
 	}
 
+	// Upsert user identities
+	var ids []mysqlIdentity
+	for _, uIdentity := range u.Identities {
+		ids = append(ids, mysqlIdentity{
+			publicKey:   uIdentity.String(),
+			activated:   uIdentity.Activated,
+			deactivated: uIdentity.Deactivated,
+			userID:      ur.ID,
+		})
+	}
+	err = upsertIdentities(ctx, tx, ids)
+	if err != nil {
+		return fmt.Errorf("insert new identities: %v", err)
+	}
+
+	// Commit transaction.
+	if err := tx.Commit(); err != nil {
+		if err2 := tx.Rollback(); err2 != nil {
+			// We're in trouble!
+			panic(fmt.Errorf("rollback tx failed: commit:'%v' rollback:'%v'",
+				err, err2))
+		}
+		return fmt.Errorf("commit tx: %v", err)
+	}
+
+	return nil
+}
+
+func upsertIdentities(ctx context.Context, tx *sql.Tx, ids []mysqlIdentity) error {
+	q := "INSERT INTO identities(publicKey, userID, activated, deactivated) VALUES "
+	vals := []interface{}{}
+
+	for _, id := range ids {
+		q += "(?, ?, ?, ?),"
+		vals = append(vals, id.publicKey, id.userID, id.activated, id.deactivated)
+	}
+	// Trim the last ,
+	q = strings.TrimSuffix(q, ",")
+
+	// Update activated & deactivated columns when key already exists.
+	q += "ON DUPLICATE KEY UPDATE activated=VALUES(activated), " +
+		"deactivated=VALUES(deactivated)"
+
+	// Prepare the statement
+	stmt, err := tx.PrepareContext(ctx, q)
```

Comment:
Good question, apparently there is no advantage here in using `Prepare` instead of `Exec`.

> Prepare
This should be used only when we want to prepare once at the start of the program and execute N number of times during the course of the program.  

Which means only when using the same prepared statement multiple times and close it on program exit which
is clearly not the case with this insert statement.